src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -687,15 +686,19 @@ JvmtiEnvBase::get_current_contended_monitor(JavaThread *calling_thread, JavaThre
 jvmtiError
 JvmtiEnvBase::get_owned_monitors(JavaThread *calling_thread, JavaThread* java_thread,
                                  GrowableArray<jvmtiMonitorStackDepthInfo*> *owned_monitors_list) {
+  // Note:
+  // calling_thread is the thread that requested the list of monitors for java_thread.
+  // java_thread is thread owning the monitors.
+  // current_thread is thread executint this code, can be a non-JavaThread (e.g. VM Thread).
```

Comment:
The assumption that it must be an JavaThread was due to code such as "JavaThread::current();", any Thread can execute this.
The default the first thread grabbing the handshake state lock for the targeted thread will execute all handshake operation it is allowed to. This might be the VM thread.